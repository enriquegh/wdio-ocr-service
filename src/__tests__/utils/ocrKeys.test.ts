import ocrKeys from '../../utils/ocrKeys'

const globalAny:any = global

describe('utils - ocrKeys', () => {
  it('should call the performActions with the correct keys', async () => {
    globalAny.driver = {
      performActions: jest.fn()
    }

    await ocrKeys('123')

    expect(globalAny.driver.performActions).toHaveBeenCalledWithSnapshot()
  })
})
