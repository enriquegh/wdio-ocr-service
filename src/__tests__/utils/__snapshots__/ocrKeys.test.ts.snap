// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils - ocrKeys should call the performActions with the correct keys 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "actions": Array [
            Object {
              "type": "keyDown",
              "value": "1",
            },
            Object {
              "type": "keyUp",
              "value": "1",
            },
            Object {
              "type": "keyDown",
              "value": "2",
            },
            Object {
              "type": "keyUp",
              "value": "2",
            },
            Object {
              "type": "keyDown",
              "value": "3",
            },
            Object {
              "type": "keyUp",
              "value": "3",
            },
          ],
          "id": "keyboard",
          "type": "key",
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
